Makefile

```diff
@@ -138,7 +138,12 @@ CXX_FLAGS += -funwind-tables
 print-%:
 	@echo '$*=$($*)'
 
-LLVM_STATIC_LIBS = -L $(LLVM_LIBDIR) $(shell $(LLVM_CONFIG) --libs bitwriter bitreader linker ipo mcjit $(X86_LLVM_CONFIG_LIB) $(ARM_LLVM_CONFIG_LIB) $(OPENCL_LLVM_CONFIG_LIB) $(METAL_LLVM_CONFIG_LIB) $(PTX_LLVM_CONFIG_LIB) $(AARCH64_LLVM_CONFIG_LIB) $(MIPS_LLVM_CONFIG_LIB) $(POWERPC_LLVM_CONFIG_LIB) $(HEXAGON_LLVM_CONFIG_LIB))
+# Append the --link-static flag to llvm-config if it exists. We can
+# make this unconditional once llvm 4.0 is the minimum version we
+# support.
+LLVM_LINK_STATIC_FLAG = $(shell $(LLVM_CONFIG) --link-static 2>/dev/null && echo --link-static)
```

Comment:
"echo --link-static" will probably work everywhere, but it is syntactically passing a command line flag to echo and thus relies on echo not treating switches it does not recognize as plain arguments. Given the shape of the world, I'd rather avoid this. Only solution I can think though is to put it in quotes and prepend a space: " --link-static". Could also have it print nothing and use another make conditional variable assignment to set the same var to --link-static.